MOVNTI is used to clear pagetables and is a good hint to unshadow, 
so don't bother emulating it until we run into a real-world need for it.

diff -r 29d7e3522cc5 xen/arch/x86/x86_emulate/x86_emulate.c
--- a/xen/arch/x86/x86_emulate/x86_emulate.c	Mon Jun 01 14:56:17 2009 +0100
+++ b/xen/arch/x86/x86_emulate/x86_emulate.c	Tue Jun 02 11:58:44 2009 +0100
@@ -238,7 +238,7 @@
     ByteOp|DstReg|SrcMem|ModRM|Mov, DstReg|SrcMem16|ModRM|Mov,
     /* 0xC0 - 0xC7 */
     ByteOp|DstMem|SrcReg|ModRM, DstMem|SrcReg|ModRM,
-    0, DstMem|SrcReg|ModRM|Mov,
+    0, 0, /* MOVNTI explicitly removed to help shadow code */
     0, 0, 0, ImplicitOps|ModRM,
     /* 0xC8 - 0xCF */
     ImplicitOps, ImplicitOps, ImplicitOps, ImplicitOps,
@@ -3886,11 +3886,13 @@
         }
         goto add;
 
+#if 0 /* MOVNTI explicity removed to help shadow code */
     case 0xc3: /* movnti */
         /* Ignore the non-temporal hint for now. */
         generate_exception_if(dst.bytes <= 2, EXC_UD, -1);
         dst.val = src.val;
         break;
+#endif
 
     case 0xc7: /* Grp9 (cmpxchg8b/cmpxchg16b) */ {
         unsigned long old[2], exp[2], new[2];
